src/ConnectedMode.UnitTests/Migration/VsAwareFileSystemTests.cs

```diff
@@ -18,11 +18,17 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
```

Comment:
We use threadhandling 3 times in the actual class and I think we should have some tests covering that it runs on the expected thread

Reply:
@bigfluffycookie I've added threading unit tests for the two cases where we use the VS interface (the calls that crash VS if we are not on the UI thread).

Reply:
(and I fixed the code smell)